alphagov/paas-product-pages
# Dynamic page titles

## What
Restructure page layouts and export page metadata so that we can properly set `title` meta value to `<pageName> - <siteName>`

To do this, we now use `config.json` to set the `siteName`.
An export of metadata was added to each `.mdx` page and passed on to the layout the page uses so that we can update the `title` property there as well as the main page heading

As for the layouts, `PageWithSideNav` was moved into it's own folder and rename to index for easier imports (and future tests). 

A GenericLayout was added for pages that don't below to a section and don't have a section nav.

As a result, `title` meta and content HTML structure was removed from _document.js, also fixes
`Warning: <title> should not be used in _document.js's <Head>. https://err.sh/next.js/no-document-title` error

## Changes

### Before
<img width="1104" alt="Screenshot 2020-06-04 at 09 16 31" src="https://user-images.githubusercontent.com/3758555/83732591-2f504c80-a644-11ea-96bc-a7d516542fa6.png">


### After
<img width="1099" alt="Screenshot 2020-06-04 at 09 14 35" src="https://user-images.githubusercontent.com/3758555/83732415-e8625700-a643-11ea-9385-6aaec5f4dba5.png">

## How to review

- clone project
- checkout the `dynamic-page-titles` branch
- optional: install node if you don't have it already
- install dependencies with `npm install`
- run `npm run dev` to build the app
- go to `http://localhost:3000/` to check the tab title - should be "GOV.UK Platform as a Service"
- go to `http://localhost:3000/get-started` to check the tab title - should be ""GOV.UK Platform as a Service - Get started with GOV.UK PaaS"
- review the code for any improvements
